.github/actions/maven-build-and-tag/action.yml

```diff
@@ -81,13 +81,6 @@ runs:
   steps:
     - uses: actions/checkout@v4
 
-    - uses: actions/cache@v3
```

Comment:
I think this should be kept, just made dependent on whether we're handling a PR or a push: if it's a push we keep things as is, if it's a PR we only restore cache with the given restore keys, instead of also saving it at the end of the build

Reply:
this entry makes no sense here, because the same command is located in setup-jva

Reply:
we're not calling the local `./actions/setup-java-build` here, but the regular `actions/setup-java`, and the latter should not handle cache at all, that was the purpose of https://github.com/Alfresco/alfresco-build-tools/pull/147

Reply:
🤦‍♂️ 
Ok, that changes a lot ! sorry for that

Reply:
no worries :+1: